frontend/src/metabase-lib/query.ts

```diff
@@ -122,8 +123,9 @@ export function canSave(query: Query, cardType: CardType): boolean {
 export function asReturned(
```

Comment:
FIXME. This is hacked because this function is only used for "new expression" stuff in chill mode. Introduce a new function with better semantics.

Reply:
It's not clear to me yet what the problem is here. Is it that the new callers for this don't really want `asReturned`, they want `prepare-query`? (ie. wrapped with an MBQL stage?) That could be exposed to JS.

I'll have to play around with this PR.